python/pyspark/serializers.py

```diff
@@ -199,6 +211,46 @@ def __repr__(self):
         return "ArrowSerializer"
 
 
+class ArrowPandasSerializer(ArrowSerializer):
+    """
+    Serializes Pandas.Series as Arrow data.
+    """
+
+    def __init__(self):
+        super(ArrowPandasSerializer, self).__init__()
+
+    def dumps(self, series):
+        """
+        Make an ArrowRecordBatch from a Pandas Series and serialize. Input is a single series or
+        a list of series accompanied by an optional pyarrow type to coerce the data to.
+        """
+        import pyarrow as pa
+        # Make input conform to [(series1, type1), (series2, type2), ...]
+        if not isinstance(series, (list, tuple)) or \
+                (len(series) == 2 and isinstance(series[1], pa.DataType)):
+            series = [series]
+        series = [(s, None) if not isinstance(s, (list, tuple)) else s for s in series]
```

Comment:
I'd use generator comprehension.

Reply:
That would work, but does it help much since `series` will already be a list or tuple?

Reply:
Yea, it actually affects the performance because we can avoid an extra loop:

```python
def im_map(x):
    print("I am map %s" % x)
    return x

def im_gen(x):
    print("I am gen %s" % x)
    return x

def im_list(x):
    print("I am list %s" % x)
    return x

items = list(range(3))
map(im_map, [im_list(item) for item in items])
map(im_map, (im_gen(item) for item in items))
```

And .. this actually affects the performance up to my knowledge:

```python
import time

items = list(xrange(int(1e8)))

for _ in xrange(10):
    s = time.time()
    _ = map(lambda x: x, [item for item in items])
    print "I am list comprehension with a list: %s" % (time.time() - s)
    s = time.time()
    _ = map(lambda x: x, (item for item in items))
    print "I am generator expression with a list: %s" % (time.time() - s)
```

This gives me ~13% improvement :).

Reply:
This might not be a big deal but .. I usually use generator if it iterates once and is discarded. This should consume less memory too as list comprehension should be evaluated once first.

Reply:
Thanks @HyukjinKwon , I suppose if there are more than a few series then it might make some difference.  In that case, every little bit helps so sounds good to me!